### PR TITLE
ARO 3.11 CI to build farm

### PR DESCRIPTION
**Pull request by mjudeikis**
Move ARO CI back to build farms
Move image mirroring to standard ways

### CODE REVIEW

File: ci-operator/jobs/infra-image-mirroring.yaml [hunk 1]
```diff
@@ -95,6 +95,54 @@ periodics:
...
+      - mountPath: /home/mirror/.docker/config.json
+        name: push
+        readOnly: true
+        subPath: config.json
+      - mountPath: /etc/imagemirror
+        name: config
+    volumes:
+    - name: push
+      secret:
+        secretName: registry-push-credentials-quay.io-openshift-azure
```
Comment (hongkailiu) [inline ci-operator/jobs/infra-image-mirroring.yaml:142]: /hold

Let me create that secret.

File: core-services/sanitize-prow-jobs/_config.yaml [hunk 2]
```diff
@@ -1629,139 +1629,6 @@ groups:
...
-    - pull-ci-azure-release-v20-unit
-    - pull-ci-azure-release-v20-upgrade-private-v15.0
-    - pull-ci-azure-release-v20-upgrade-private-v16.1
-    - pull-ci-azure-release-v20-upgrade-private-v17.0
-    - pull-ci-azure-release-v20-upgrade-private-v19.0
-    - pull-ci-azure-release-v20-upgrade-v15.0
-    - pull-ci-azure-release-v20-upgrade-v16.1
-    - pull-ci-azure-release-v20-upgrade-v17.0
-    - pull-ci-azure-release-v20-upgrade-v19.0
-    - branch-ci-openshift-release-master-release-controller-annotate
```
Comment (hongkailiu) [inline core-services/sanitize-prow-jobs/_config.yaml:1761]: From this job, they are not azure jobs, right?

File: core-services/sanitize-prow-jobs/_config.yaml [hunk 3]
```diff
@@ -1629,139 +1629,6 @@ groups:
     paths:
     - ".*-postsubmits.yaml$"
   "api.ci":
-    jobs:
-    - periodic-ci-azure-conformance
```
Comment (hongkailiu) [inline core-services/sanitize-prow-jobs/_config.yaml:1633]: I would separate moving jobs with another PR also start to test one job only.

Comment (mjudeikis) [reply core-services/sanitize-prow-jobs/_config.yaml:1633]: We currently don't have any activity in this area. If worry being multiple failures -ok, If interruption - we don't mind this at this point.  Your call.

File: core-services/sanitize-prow-jobs/_config.yaml (hunk 2 above)
Comment (mjudeikis) [reply core-services/sanitize-prow-jobs/_config.yaml:1761]: rebase error. will fix

File: ci-operator/jobs/infra-image-mirroring.yaml (hunk 1 above)
Comment (hongkailiu) [reply ci-operator/jobs/infra-image-mirroring.yaml:142]: Done!

```
oc --context app.ci get secret registry-push-credentials-quay.io-openshift-azure -n ci
NAME                                                TYPE                             DATA   AGE
registry-push-credentials-quay.io-openshift-azure   kubernetes.io/dockerconfigjson   1      63s
```

File: ci-operator/jobs/infra-image-mirroring.yaml [hunk 4]
```diff
@@ -95,6 +95,54 @@ periodics:
...
+    ci.openshift.io/role: image-mirroring
+  name: periodic-image-mirroring-openshift-azure
+  spec:
+    containers:
+    - args:
+      - -c
+      - |
+        set -o errexit
+        failures=0
+        for mapping in /etc/imagemirror/mapping_open_cluster_management*; do
```
Comment (hongkailiu) [inline ci-operator/jobs/infra-image-mirroring.yaml:113]: `mapping_openshift_azure*`

File: ci-operator/jobs/infra-image-mirroring.yaml [hunk 5]
```diff
@@ -95,6 +95,54 @@ periodics:
...
+          echo "Running: oc image mirror -f=$mapping --skip-multiple-scopes"
+          if ! oc image mirror -f="$mapping" --skip-multiple-scopes; then
```
Comment (hongkailiu) [inline ci-operator/jobs/infra-image-mirroring.yaml:115]: We need oc-cli here have access to the source images.
Do you mind open the image on app.ci to the public?
This is an example: https://github.com/openshift/release/blob/309118783cd6dc0422ddc8a88d2c786a6ce60105/clusters/app.ci/ocp-kni/admin_ocp-kni_list.yaml#L8

Otherwise, we have to oc-login or provide a pull token in some way.

File: core-services/sanitize-prow-jobs/_config.yaml (hunk 3 above)
Comment (hongkailiu) [reply core-services/sanitize-prow-jobs/_config.yaml]: I would separate.
Easy to revert upon issues. :)

**Comment by hongkailiu** [comment]
Could we rename?

`clusters/app.ci/openshift-azure/admn_openshift-azure_list.yaml`

to 

`clusters/app.ci/azure/admin_openshift-azure_list.yaml`

**Review by hongkailiu — APPROVED** [review]
